File: AdvancePython/Day-2 Constructors and Methods/Task-3-4.py
# Task 3:
# 4. Accessors:
class Car:
    def __init__(self, brand: str) -> None:
        self._brand: str = brand  # '_(underscore prefix)' means 'internal use'

    @property
    def brand_name(self) -> str:
        return self._brand.upper()


car_brand = Car("BMW")
print(f"Car brand is: {car_brand.brand_name}")

File: node.js/Day-2/Task-3-4.js
// Task 3:
// 4. Accessors (Getters):

class Car {
    constructor(brand) {
        this._brand = brand
    }

    get brand_name() {
        return this._brand.toUpperCase();
    }
}

const car = new Car("Audi");
console.log(car.brand_name);